lib/saml_idp/encryptor.rb

```diff
@@ -34,8 +34,25 @@ def openssl_cert
     end
     private :openssl_cert
 
+    # Encryption algorithms enumerated in:
+    # https://github.com/digidentity/xmlenc/blob/937ca2f/lib/xmlenc/encrypted_data.rb#L3-L10
     def block_encryption_ns
-      "http://www.w3.org/2001/04/xmlenc##{block_encryption}"
+      case block_encryption
+        when 'tripledes-cbc'
+          'http://www.w3.org/2001/04/xmlenc#tripledes-cbc'
+        when 'aes128-cbc'
+        'http://www.w3.org/2001/04/xmlenc#aes128-cbc'
+        when 'aes256-cbc'
+          'http://www.w3.org/2001/04/xmlenc#aes256-cbc'
+        when 'aes128-gcm'
+          'http://www.w3.org/2009/xmlenc11#aes128-gcm'
+        when 'aes192-gcm'
+          'http://www.w3.org/2009/xmlenc11#aes192-gcm'
+        when 'aes256-gcm'
+          'http://www.w3.org/2009/xmlenc11#aes256-gcm'
+        else
+          "http://www.w3.org/2009/xmlenc11##{block_encryption}"
```

Comment:
should we have an `else` here? or should we throw if it's not an expected encryption algorithm? (the block_encryption value is derived from the service provider in the database)

Reply:
I think we definitely could, `xmlenc` will throw anyway:

```
Xmlenc::UnsupportedError - Unsupported encryption method http://www.w3.org/2001/04/xmlenc#abc-haha
```

Reply:
Changed to raise a `ValidationError`